crates/ra_hir/src/ty/infer.rs

```diff
@@ -967,8 +988,51 @@ impl<'a, D: HirDatabase> InferenceContext<'a, D> {
                 Ty::unit()
             }
             Expr::For { iterable, body, pat } => {
-                let _iterable_ty = self.infer_expr(*iterable, &Expectation::none());
-                self.infer_pat(*pat, &Ty::Unknown, BindingMode::default());
+                use crate::nameres::PerNs;
+                use crate::resolve::Resolution::Def;
+                use crate::code_model::ModuleDef::Trait;
+                use super::traits::ProjectionPredicate;
+                use super::ProjectionTy;
+
+                let iterable_ty = self.infer_expr(*iterable, &Expectation::none());
+
+                let into_iter_path = Path {
+                    kind: PathKind::Plain,
```

Comment:
this should probably be `PathKind::Abs`, actually, I think

Reply:
I tried that, but couldn’t find a way to create a test with `std` crate so I changed to Plain. Any idea?

Reply:
oh, right. [Here](https://github.com/rust-analyzer/rust-analyzer/blob/master/crates/ra_hir/src/ty/tests.rs#L2557)'s a test that constructs a crate graph; the same approach should work for this.